src/features/machine/PolarMachine.js

```diff
@@ -88,13 +88,13 @@ export default class PolarMachine extends Machine {
   }
 
   // Returns whether a given path lies on the perimeter of the circle.
-  onPerimeter(v1, v2, delta=.001) {
+  onPerimeter(v1, v2, delta=1) {
     let rm = Math.pow(this.settings.maxRadius, 2)
     let r1 = Math.pow(v1.x, 2) + Math.pow(v1.y, 2)
     let r2 = Math.pow(v2.x, 2) + Math.pow(v2.y, 2)
     let d = this.perimeterDistance(v1, v2)
 
-    return (r1 >= rm - delta && r2 >= rm - delta) && d < 15
+    return Math.abs(r1 - rm) < delta && Math.abs(r2 - rm) < delta && d < 4*Math.PI
```

Comment:
What is the case where d would be greater than 4pi? I'm not sure why there is even a permiterDistance involved in this function.

Reply:
Oh, wait, this is trying to determine if the vector from v1 to v2 is along the perimeter (already, not assuming it's following the arc).

So both points need to be near the max radius, and the distance between them needs to be small enough to be approximated by a line or an arc. So -maxRadius,0 to maxRadius,0 would be too far to be approximated by a line.

4pi is pretty close to 15, so I guess it didn't change much. I feel like there should be a /radius in there somewhere.

Reply:
If you are trying to limit it to a specific angle (in radians) it would be:

d / (2pi*r) < (angle/2pi)
d < angle * maxRadius

A specific arc length would also make some sense. 

If you wanted to determine the max distance between the arc and the connecting line, it would happen halfway between the two points, and it would be the difference between the long side and the maxRadius.

Reply:
That looks good, but it is totally garbage.. Not sure what the problem is. Maybe the small angle approx. is removing too much of the error. I have to put in pretty crazy numbers to get 15 out of it.

Reply:
Without the angle approx:
error = maxRadius(1-cos(d/2maxRadius))
d < acos(1-maxError/maxRadius)*2maxRadius

If maxError is 0.1 and maxRadius is 400, that comes up with a max d of 17.9, so that is reasonable.

FTR, I don't really care this much :). I think it is splitting hairs for drawing in sand. It is an interestin problem to think about though.

Reply:
Ok. I tried to document my rationale for the logic used. The case to test, if you are interested, is a Polygon, 4 sides, with a default spin (polar machine). If you don't keep the large delta, as you grow the shape, you'll continue to add distance (it never stabilizes). If you don't add the max. distance check, you'll drop valid polygon lines near the perimeter. Hopefully my inline comments make sense.